### ISSUE TITLE
Core algorithm

### ISSUE THREAD

**Issue by sekunho** [butlerph/scheduler#1]
The core algorithm should be completed before additional optimizations.

## Operators

- Crossover
  - [x] Horizontal substring
  - [ ] Vertical substring (Nice to have)
- [x] Mutation
- Repair
  - [x] Remove duplicates
  - [x] Remove todos to prevent excess duration 
  - [x] Reassign unassigned todos in FF-D
